MVJ-550 override_receivable_type in Rent (UI)

juho-kettunen-nc opened the pull request:
This implements the UI part of override receivable type, where it is a property of **Rent**, instead of ContractRent as in https://github.com/City-of-Helsinki/mvj-ui/pull/544

juho-kettunen-nc commented:
Still requires customer blessing to use this Rent version instead of the ContractRent, but this is the best version of the logic we have right now.

henrinie-nc commented on src/leases/helpers.ts:
```diff
@@ -2948,4 +2951,15 @@ export const sortRelatedHistoryItems = (a: Record<string, any>, b: Record<string
+ * @param leaseServiceUnitId ID of the lease's service unit
+ * @returns boolean Whether the field should be visible
+ */
+export const showOverrideReceivableTypeField = (leaseServiceUnitId: number): boolean => {
```
If serviceunit has a type, the type of the argument here could be ServiceUnit["id"], which would end up with the same type number, but maybe providing more context

juho-kettunen-nc replied:
Ah, didn't know that's possible

juho-kettunen-nc replied:
Not relevant anymore because the function was removed

henrinie-nc reviewed (approved):
This seems usable. Improvement might be to add an attribute to the service unit model to dictate whether to use override field for this service unit, so we would not need to hard code lists of ID's and define code with service unit names.

Check my comments on the changes before merging.

juho-kettunen-nc commented on src/leases/components/LeasePage.tsx:
```diff
@@ -261,6 +266,7 @@ class LeasePage extends Component<Props, State> {
 
     if (prevProps.currentLease !== currentLease) {
       this.setPageTitle();
+      fetchReceivableTypes();
```
No, because, these receivable types are only related to the lease's receivable type. The set of available receivable types should not change depending on the active service unit.

juho-kettunen-nc commented on src/leases/components/leaseSections/rent/BasicInfoEdit.tsx:
```diff
@@ -534,6 +596,7 @@ type Props = {
   isSaveClicked: boolean;
   leaseAttributes: Attributes;
   leaseTypes: LeaseTypeList;
+  receivableTypes: Array<Record<string, any>>;
```
Yes

juho-kettunen-nc replied:
Added in 96e3d70d93757cbcfcc625aa887d0e8d76c6d515

juho-kettunen-nc commented:
Added new field `use_rent_override_receivable_type` to the ServiceUnit model, and adjusted logic accordingly.
- API https://github.com/City-of-Helsinki/mvj/pull/772/commits/1277eed1f0152ecfb9f1f422d6a8a3adfeaa8355
- UI https://github.com/City-of-Helsinki/mvj-ui/pull/545/commits/fb1f3f31c8652e6b3a050104ed07587d06e47bf3

Input validation and tests in https://github.com/City-of-Helsinki/mvj/pull/772/commits/9e4c04186fca606be7cff99c9d3bdeda5e26edac

henrinie-nc commented:
> Added new field `use_rent_override_receivable_type` to the ServiceUnit model, and adjusted logic accordingly.
> 
>     * API [City-of-Helsinki/mvj@1277eed](https://github.com/City-of-Helsinki/mvj/commit/1277eed1f0152ecfb9f1f422d6a8a3adfeaa8355)
> 
>     * UI [fb1f3f3](https://github.com/City-of-Helsinki/mvj-ui/commit/fb1f3f31c8652e6b3a050104ed07587d06e47bf3)
> 
> 
> Input validation and tests in [City-of-Helsinki/mvj@9e4c041](https://github.com/City-of-Helsinki/mvj/commit/9e4c04186fca606be7cff99c9d3bdeda5e26edac)

Looks good for the UI part!

juho-kettunen-nc commented:
Reminder to self: squash, or rebase locally
